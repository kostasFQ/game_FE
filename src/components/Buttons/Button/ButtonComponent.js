import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ onClick, title, ...rest }) => <button onClick={onClick} {...rest}>{title}</button>;

Button.defaultProps = {
  onClick: () => { },
};

Button.propTypes = {
  onClick: PropTypes.func,
  title: PropTypes.string.isRequired
}

export default Button;