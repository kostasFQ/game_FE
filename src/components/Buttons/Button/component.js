import React from 'react';

const Button = ({ onClick, title, ...rest }) => (
  <button onClick={onClick} { ...rest }>{ title }</button>
);

export default Button;