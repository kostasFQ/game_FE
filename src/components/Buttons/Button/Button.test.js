import React from 'react';
import Button from './component';
import renderer from 'react-test-renderer';

test('Button should be render', async () => {
  const button = await renderer.create(
    <Button title='Start' disabled={true} onClick={ () => {} } />
  );
  let component = button.toJSON();
  expect(component).toMatchSnapshot();
})