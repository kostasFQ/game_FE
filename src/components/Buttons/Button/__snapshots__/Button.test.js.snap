// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should be render 1`] = `
<button
  disabled={true}
  onClick={[Function]}
>
  Start
</button>
`;
